examples/tests/evm-price-feed.test.ts

```diff
@@ -0,0 +1,171 @@
+// biome-ignore assist/source/organizeImports: biome is lying
+import { file } from 'bun';
+import { afterEach, describe, expect, it, mock } from 'bun:test';
+import { testOracleProgramExecution, testOracleProgramTally } from '@seda-protocol/dev-tools';
+import { match } from 'ts-pattern';
+import {
+  handleBigIntArrayTallyVmResult as handleTallyVmResult,
+  createSuccessfulJsonBigIntArrayReveal as createSuccessfulReveal,
+  createFailedReveal,
+} from './utils.js';
+import { ethers } from 'ethers';
+
+const WASM_PATH = 'target/wasm32-wasip1/release/evm-price-feed.wasm';
+
+const fetchMock = mock();
+
+afterEach(() => {
+  fetchMock.mockRestore();
+});
+
+describe('evm price feed', () => {
+  it('should return the correct price', async () => {
+    fetchMock.mockImplementation((url) => {
+      return match(url.host)
+        .with('api.binance.com', () => {
+          return new Response(JSON.stringify({ symbol: 'BTCUSDT', price: '117318.90000000' }));
+        })
+        .otherwise(() => {
+          throw new Error(`Unexpected host: ${url.host}`);
+        });
+    });
+
+    const oracleProgram = await file(WASM_PATH).arrayBuffer();
+    const tickers = ["BTC-USDT"];
+    const execInputsStr = ethers.AbiCoder.defaultAbiCoder().encode(
+      ["string[]"],
+      [tickers]
+    );
+    // Remove '0x' prefix
+    const execInputs = Buffer.from(execInputsStr.slice(2), 'hex');
+
+    const vmResult = await testOracleProgramExecution(
+      Buffer.from(oracleProgram),
+      execInputs, 
+      fetchMock
+    );
+
+    const jsonString = Buffer.from(vmResult.result).toString('utf-8');
```

Comment:
We also have a `handleJsonBigIntExecutionVmResult` which you can make another variant of to handle an array of JSON BigInts. This way we can re-use that function in the future/even in your later test instead of having duplicated code.